src/Spout/Reader/ODS/RowIterator.php

```diff
@@ -130,23 +139,23 @@ public function next()
                 } else if ($this->xmlReader->isPositionedOnEndingNode(self::XML_NODE_ROW)) {
                     // End of the row description
                     $isEmptyRow = ($numCellsRead <= 1 && $this->isEmptyCellValue($cellValue));
-                    if ($isEmptyRow) {
-                        // skip empty rows
-                        $this->next();
-                        return;
-                    }
 
-                    // Only add the value if the last read cell is not a trailing empty cell repeater in Excel.
-                    // The current count of read columns is determined by counting the values in $rowData.
-                    // This is to avoid creating a lot of empty cells, as Excel adds a last empty "<table:table-cell>"
-                    // with a number-columns-repeated value equals to the number of (supported columns - used columns).
-                    // In Excel, the number of supported columns is 16384, but we don't want to returns rows with
-                    // always 16384 cells.
-                    if ((count($rowData) + $numColumnsRepeated) !== self::MAX_COLUMNS_EXCEL) {
-                        for ($i = 0; $i < $numColumnsRepeated; $i++) {
-                            $rowData[] = $cellValue;
+                    if (!$isEmptyRow) {
+                        // Only add the value if the last read cell is not a trailing empty cell repeater in Excel.
+                        // The current count of read columns is determined by counting the values in $rowData.
+                        // This is to avoid creating a lot of empty cells, as Excel adds a last empty "<table:table-cell>"
+                        // with a number-columns-repeated value equals to the number of (supported columns - used columns).
+                        // In Excel, the number of supported columns is 16384, but we don't want to returns rows with
+                        // always 16384 cells.
+                        if ((count($rowData) + $numColumnsRepeated) !== self::MAX_COLUMNS_EXCEL) {
+                            for ($i = 0; $i < $numColumnsRepeated; $i++) {
+                                $rowData[] = $cellValue;
+                            }
                         }
-                        $this->numReadRows++;
+                    } elseif ($this->readerOptions->shouldPreserveEmptyRows()) {
+                        $rowData = array_fill(0, $numColumnsRepeated, '');
```

Comment:
This will create the "a lot of empty cells". For example if you take the test ```ODS\ReaderTest::testReadWithFilesGeneratedByExternalSoftwares``` and insert:

```    
public function testReadWithFilesGeneratedByExternalSoftwares($fileName, $skipLastEmptyValues)
{
        $reader = ReaderFactory::create(Type::ODS);
        /** @var  $reader Reader */
        $reader->setShouldPreserveEmptyRows(true);

        $allRows = $this->getAllRowsForFile($fileName, $reader);
 // ....
```

you will get 16384 empty cells in the "preserved" empty row.  I am not sure how to circumvent this - but this does not sound "right" ;)